### PR TITLE
KE-925 started with scaffold for a Context Model

### PR DESCRIPTION
**Pull request by jberends**
* [X] make crud methods in the client
* [X] make curd methods in the scope
* [X] ensure that context_type is either location/time and enable/disable use of various attributes.
* [X] write tests

Fixes #<issue>

## Checklist:
- [X] My code follows the pykechain style
    - [X] I added type hinting to all functions
    - [X] I added documentation for all public functions
    - [X] I locally used `tox` to check for dists and docs errors
    - [X] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [X] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [X] I used [`black`](https://github.com/psf/black) to format the new code
- [X] I have added tests that prove my fix is effective or that my feature works
    - [X] I committed fresh test cassettes
    - [X] I have proper test coverage (don't decline the coverage of the test)
- [X] I asked another teammate to review the code
- [X] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

**Comment by jberends** [comment]
@raduiordache can you please review

File: pykechain/defaults.py [hunk 1]
```diff
@@ -177,5 +184,9 @@
         ['id', 'subject', 'status', 'message', 'team', 'created_at', 'options', 'updated_at'])},
     'expiring_downloads': {'fields': ",".join(
         ['id', 'created_at', 'updated_at', 'expires_at', 'expires_in', 'content'])},
+    'context': {'fields': ",".join(
+        ['id', 'name', 'ref', 'created_at', 'updated_at', 'description', 'tags', 'context_type',
+         'activities', 'options', 'feature_collection', 'start_date', 'due_date']
```
Comment (raduiordache) [inline pykechain/defaults.py:189]: Shouldn't we have also the `contexts` specified here?

**Review by raduiordache — CHANGES_REQUESTED** [review]

File: tests/test_contexts.py [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+        self.assertFalse(self.context.activities)
+        self.context.link_activities(activities=[self.project.activity(name="Specify wheel diameter"), self.project.activity(name="Task - Form")])
+        self.assertEqual(len(self.context.activities), 2)
+        self.context.unlink_activities(activities=[self.project.activity(name="Specify wheel diameter")])
+        self.assertEqual(len(self.context.activities), 1)
+        self.assertListEqual(list(self.context.activities), [self.project.activity(name="Task - Form").id])
+
+
+
+#     def test_retrieve_services_with_kwargs(self):
```
Comment (raduiordache) [inline tests/test_contexts.py:108]: Please remove the uncommented lines

**Review by raduiordache — CHANGES_REQUESTED** [review]

File: pykechain/defaults.py (hunk 1 above)
Comment (jberends) [reply pykechain/defaults.py:189]: Unsure. I thought the context and contexts field were the same.

Comment (jberends) [reply pykechain/defaults.py]: sharp, corrected it - a plural for the query params _is_ needed.
In the mean time I saw that I forgot to add 'scope' to the list of fields to request. Nice!

File: tests/test_contexts.py (hunk 2 above)
Comment (jberends) [reply tests/test_contexts.py]: done

File: pykechain/defaults.py (hunk 1 above)
Comment (jberends) [reply pykechain/defaults.py]: done

**Pull request merged**